### DISCUSSION TITLE
Generate refined mesh with elements as the center

### DISCUSSION THREAD

**Discussion by BoZeng1997** [idaholab/moose#18802] [Q&A Meshing] [answered]
Hi,

I have a square domain and it is symmetric about the x-axis. I need refined mesh around the x-axis but I still want to keep a line of elements as the center, not nodes. Is there a way to achieve that in moose? Simple refinement would always break central elements. 

Bo

**Comment by GiudGiud** [comment, marked as answer]
Hi Bo

There is. You should define the center element to be in a different block then refine by block. This can be done easily in this PR: https://github.com/idaholab/moose/pull/18744

But in the mean time, you can use adaptivity to do it like in the example we are adding in this PR:
https://github.com/idaholab/moose/pull/18657

Guillaume

**Reply by BoZeng1997** [reply]
I think I did not make myself clear. I want to have odd number of elements in the y direction and they are dense mesh. The usual refinement will always split the initial elements and make a line of nodes on the x-axis. I do not care if I can keep the exact original central elements, I just want a line of refined element on the x-axis. The examples provided do not solve this issue.

**Reply by GiudGiud** [reply]
well we can’t split elements in 3 parts in one direction. So unless I’m missing something this isn’t possible through refinement

but you could just change the ny in a RectiliearMeshGenerator and refine that way while keeping an element on the x axis

**Reply by YaqiWang** [reply]
If only quad in 2D or cube in 3D is needed, you can try `CartesianMeshGenerator`, which has more flexibility on controlling the element sizing than `GeneratedMeshGenerator`.
